src/env.rs

```diff
@@ -46,10 +47,10 @@ pub fn get_remaining_points_for_instance(instance: &Instance) -> u64 {
     }
 }
 
-pub fn sub_remaining_point(env: &Env, points: u64) -> anyhow::Result<()> {
+pub fn sub_remaining_point(env: &Env, points: u64) {
     let instance = &env.instance.clone().unwrap();
-    set_remaining_points(instance, get_remaining_points_for_env(env) - points);
-    Ok(())
+    let remaining_points = get_remaining_points_for_env(env);
+    set_remaining_points(instance, remaining_points.saturating_sub(points));
```

Comment:
I think this implies:
* remaining_points = 10, apply last operation with cost 100, no more instructions afterwards => does the 100-cost op go through ?  And if it doesn't, does it imply that if the last operation's cost was exactly 10 it would not have gone through either ?